ClosedXML/Excel/Ranges/XLRangeAddress.cs

```diff
@@ -113,34 +120,76 @@ IXLAddress IXLRangeAddress.FirstAddress
         {
             [DebuggerStepThrough]
             get { return FirstAddress; }
-            set
-            {
-                if (value is XLAddress)
-                    FirstAddress = (XLAddress)value;
-                else
-                    FirstAddress = new XLAddress(value.RowNumber, value.ColumnNumber, value.FixedRow, value.FixedColumn);
-            }
         }
 
         IXLAddress IXLRangeAddress.LastAddress
         {
             [DebuggerStepThrough]
             get { return LastAddress; }
-            set
+        }
+
+        public bool IsValid
+        {
+            get
             {
-                if (value is XLAddress)
-                    LastAddress = (XLAddress)value;
-                else
-                    LastAddress = new XLAddress(value.RowNumber, value.ColumnNumber, value.FixedRow, value.FixedColumn);
+                return _firstAddress.IsValid &&
+                       _lastAddress.IsValid;
             }
         }
 
-        public bool IsValid { get; set; } = true;
-
         #endregion Public properties
 
         #region Public methods
 
+        /// <summary>
+        /// Lead a range address to a normal form - when <see cref="FirstAddress"/> points to the top-left address and
+        /// <see cref="LastAddress"/> points to the bottom-right address.
+        /// </summary>
+        /// <returns></returns>
+        public XLRangeAddress Normalize()
```

Comment:
Can this method not be simplified with `Min` and `Max` functions?

Reply:
I doubt so. Depending on which row (column) is used for `firstRow` (`firstColumn`) we use different values for `FixedRow` (`FixedColumn`). So `Min` / `Max` would not help very much.